readthedocs/search/parsers.py

```diff
@@ -140,21 +140,68 @@ def _parse_sections(self, title, body):
         except Exception as e:
             log.info('Unable to index section', section=str(e))
 
+        # All terms in dls are treated as sections.
+        # After indexing them, we remove them from the body such that their contents aren't
+        # repeated in the following indexing
+        dls = body.css("dl")
+        dl_nodes_to_be_removed = []
+        for dl in dls:
+            # Select all dts with id defined
+            dts = dl.css('dt[id]:not([id=""])')
+
+            # https://developer.mozilla.org/en-US/docs/Web/HTML/Element/dt
+            # multiple <dt> elements in a row indicate several terms that are
+            # all defined by the immediate next <dd> element.
+            for dt in dts:
+                if dl not in dl_nodes_to_be_removed:
+                    dl_nodes_to_be_removed.append(dl)
+                try:
+                    title, _id = self._parse_dt(dt)
+                    # https://developer.mozilla.org/en-US/docs/Web/CSS/General_sibling_combinator
+                    dd = dt.css_first("dt ~ dd")
+                    # We only index a dt with an id attribute and an accompanying dd
+                    if not dd or not _id:
+                        continue
+
+                    # The content of the <dt> section is the content of the accompanying <dd>
+                    content = dd.text()
+                    yield {
+                        "id": _id,
+                        "title": title,
+                        "content": content,
+                    }
+                except Exception as e:
+                    log.info("Unable to index dt section.", section=str(e))
+
+        # Remove the <dl> nodes that were found to have indexed content
+        for node in dl_nodes_to_be_removed:
+            node.decompose()
+
         # Index content from h1 to h6 headers.
         for head_level in range(1, 7):
-            tags = body.css(f'h{head_level}')
+            tags = body.css(f"h{head_level}")
             for tag in tags:
                 try:
-                    title, id = self._parse_section_title(tag)
+                    title, _id = self._parse_section_title(tag)
                     next_tag = self._get_header_container(tag).next
                     content, _ = self._parse_section_content(next_tag, depth=2)
                     yield {
-                        'id': id,
-                        'title': title,
-                        'content': content,
+                        "id": _id,
+                        "title": title,
+                        "content": content,
                     }
                 except Exception as e:
-                    log.info('Unable to index section.', section=str(e))
+                    log.info("Unable to index section.", section=str(e))
```

Comment:
```suggestion
                    log.info("Unable to index section.", exc_info=True)
```

Reply:
thanks, it's probably better to print out the entire exception for sure. I was wondering if the previous pattern was there to silence a lot of anticipated exceptions, but this is really an anti-pattern.